exporter/elasticsearchexporter/model_test.go

```diff
@@ -218,6 +218,61 @@ func TestEncodeEvents(t *testing.T) {
 	}
 }
 
+func TestEncodeLogECSModeDuplication(t *testing.T) {
+	resource := pcommon.NewResource()
+	err := resource.Attributes().FromRaw(map[string]any{
+		semconv.AttributeServiceName:    "foo.bar",
+		semconv.AttributeHostName:       "localhost",
+		semconv.AttributeServiceVersion: "1.1.0",
+		semconv.AttributeOSType:         "darwin",
+		semconv.AttributeOSDescription:  "Mac OS Mojave",
+		semconv.AttributeOSName:         "Mac OS X",
+		semconv.AttributeOSVersion:      "10.14.1",
+	})
+	require.NoError(t, err)
+
+	expectedDocFields := pcommon.NewMap()
+	err = expectedDocFields.FromRaw(map[string]any{
+		"service.name":     "foo.bar",
+		"service.version":  "1.1.0",
+		"agent.name":       "otlp",
+		"host.os.platform": "darwin",
+		"host.os.full":     "Mac OS Mojave",
+		"host.os.name":     "Mac OS X",
+		"host.os.version":  "10.14.1",
+		"host.os.type":     "macos",
+		"host.name":        "localhost",
+	})
+
+	want := `{"@timestamp":"2024-03-12T20:00:41.123456789Z","agent":{"name":"otlp"},"container":{"image":{"tag":["v3.4.0"]}},"event":{"action":"user-password-change"},"host":{"hostname":"localhost","os":{"full":"Mac OS Mojave","name":"Mac OS X","platform":"darwin","type":"macos","version":"10.14.1"}},"service":{"name":"foo.bar","version":"1.1.0"}}`
+	require.NoError(t, err)
+
+	resourceContainerImageTags := resource.Attributes().PutEmptySlice(semconv.AttributeContainerImageTags)
+	err = resourceContainerImageTags.FromRaw([]any{"v3.4.0"})
+	require.NoError(t, err)
+
+	scope := pcommon.NewInstrumentationScope()
+
+	record := plog.NewLogRecord()
+	err = record.Attributes().FromRaw(map[string]any{
+		"event.name": "user-password-change",
+	})
+	require.NoError(t, err)
+	observedTimestamp := pcommon.Timestamp(1710273641123456789)
+	record.SetObservedTimestamp(observedTimestamp)
+
+	m := encodeModel{
+		mode:  MappingECS,
+		dedot: true,
+		dedup: true,
```

Comment:
How is this test testing de-duplication?  I ran it with setting `dedup: false` and it still passes.

Reply:
if you run it without a fix, resulting document is malformed. 
the fix was really in calling `Sort` or `Dedup` before `Serailize`